### ISSUE TITLE
Terraform v0.12.6 - Crash on Plan

### ISSUE THREAD

**Issue by RDK90** [hashicorp/terraform#25372]
### Terraform Version
Terraform version - v0.12.6
+ provider.google v3.27.0

### Terraform Configuration Files
```hcl
# main.tf

module "alb" {
    source = "./modules/load_balancer"

    customer = var.customer
    service = var.service
    stage = var.stage

    health_check = [module.mig_ssd.health_check, module.mig_hdd.health_check]
    managed_instance_group = [module.mig_ssd.managed_instance_group, module.mig_hdd.managed_instance_group]
    default_service = module.mig_ssd.managed_instance_group.self_link

    target_port = var.backend_target_port
    named_port = var.named_port
}

# load_balancer/application_load_balancer.tf

resource "google_compute_instance_group_named_port" "named_port" {
    for_each = toset(var.managed_instance_group)
    group=each.key
    name= format("%s-%s", each.key, "-named-port")
    port=var.named_port
}

resource "google_compute_backend_service" "load_balancer_backend" {
    for_each = toset(var.managed_instance_group)
    name = format("%s-%s", each.key, "-backend-service")
    port_name = [
        for named_port in google_compute_instance_group_named_port.named_port:
        named_port.name
    ]
    health_checks = var.health_check
    backend {
        group=each.key
    }
}
...
```

### Debug Output
terraform plan -var-file=test.tfvars
Refreshing Terraform state in-memory prior to plan...
The refreshed state will be used to calculate this plan, but will not be
persisted to local or remote state storage.

module.instance_template_hdd.module.gce-container.data.google_compute_image.coreos: Refreshing state...
module.instance_template_ssd.module.gce-container.data.google_compute_image.coreos: Refreshing state...

------------------------------------------------------------------------
panic: value is unknown

goroutine 871 [running]:
github.com/zclconf/go-cty/cty.Value.AsString(...)
        /opt/teamcity-agent/work/9e329aa031982669/pkg/mod/github.com/zclconf/go-cty@v1.0.1-0.20190708163926-19588f92a98f/cty/value_ops.go:1032
github.com/zclconf/go-cty/cty.Value.AsValueMap(0x2228c80, 0xc000cdff90, 0x1d08940, 0xc000c8c9e0, 0x2)
        /opt/teamcity-agent/work/9e329aa031982669/pkg/mod/github.com/zclconf/go-cty@v1.0.1-0.20190708163926-19588f92a98f/cty/value_ops.go:1095 +0x34e
github.com/hashicorp/terraform/terraform.evaluateResourceForEachExpressionKnown(0x2227840, 0xc00026d0e0, 0x225a540, 0xc000eccc30, 0xc0004f39f0, 0x434d6c, 0xc000345b00, 0x0, 0xc00004e570)
        /opt/teamcity-agent/work/9e329aa031982669/src/github.com/hashicorp/terraform/terraform/eval_for_each.go:84 +0x5bd
github.com/hashicorp/terraform/terraform.evaluateResourceForEachExpression(0x2227840, 0xc00026d0e0, 0x225a540, 0xc000eccc30, 0x1, 0x1, 0x0, 0x0)
        /opt/teamcity-agent/work/9e329aa031982669/src/github.com/hashicorp/terraform/terraform/eval_for_each.go:17 +0x5c
github.com/hashicorp/terraform/terraform.(*EvalWriteResourceState).Eval(0xc000260460, 0x225a540, 0xc000eccc30, 0x2, 0x2, 0xf4d2ad, 0x1b01300)
        /opt/teamcity-agent/work/9e329aa031982669/src/github.com/hashicorp/terraform/terraform/eval_state.go:432 +0x3a9
github.com/hashicorp/terraform/terraform.EvalRaw(0x21e73c0, 0xc000260460, 0x225a540, 0xc000eccc30, 0x1a840c0, 0x3200162, 0x19f0bc0, 0xc000cdfe80)
        /opt/teamcity-agent/work/9e329aa031982669/src/github.com/hashicorp/terraform/terraform/eval.go:57 +0x131
github.com/hashicorp/terraform/terraform.Eval(0x21e73c0, 0xc000260460, 0x225a540, 0xc000eccc30, 0xc000260460, 0x21e73c0, 0xc000260460, 0x0)
        /opt/teamcity-agent/work/9e329aa031982669/src/github.com/hashicorp/terraform/terraform/eval.go:35 +0x4d
github.com/hashicorp/terraform/terraform.(*Graph).walk.func1(0x1cfb740, 0xc000cde500, 0x0, 0x0, 0x0)
        /opt/teamcity-agent/work/9e329aa031982669/src/github.com/hashicorp/terraform/terraform/graph.go:90 +0xf40
github.com/hashicorp/terraform/dag.(*Walker).walkVertex(0xc000983e80, 0x1cfb740, 0xc000cde500, 0xc000a69000)
        /opt/teamcity-agent/work/9e329aa031982669/src/github.com/hashicorp/terraform/dag/walk.go:392 +0x353
created by github.com/hashicorp/terraform/dag.(*Walker).Update
        /opt/teamcity-agent/work/9e329aa031982669/src/github.com/hashicorp/terraform/dag/walk.go:314 +0xa9b

!!!!!!!!!!!!!!!!!!!!!!!!!!! TERRAFORM CRASH !!!!!!!!!!!!!!!!!!!!!!!!!!!!

Terraform crashed! This is always indicative of a bug within Terraform.
A crash log has been placed at "crash.log" relative to your current
working directory. It would be immensely helpful if you could please
report the crash with Terraform[1] so that we can fix this.

When reporting bugs, please include your terraform version. That
information is available on the first line of crash.log. You can also
get it by running 'terraform --version' on the command line.

[1]: https://github.com/hashicorp/terraform/issues

!!!!!!!!!!!!!!!!!!!!!!!!!!! TERRAFORM CRASH !!!!!!!!!!!!!!!!!!!!!!!!!!!!

### Crash Output
https://gist.github.com/RDK90/ecb042af3fba81b073ca8ec00f1340bc

### Expected Behavior
Standard Plan output for 2 google_compute_instance_group_named_port resources and 2 google_compute_backend_service resources

### Actual Behavior
Debug ouput as shown above

### Steps to Reproduce

This error occurs only when the output of the for_each is set to a toset method. If I pass it in as a variable or wrap it in a tomap method I get an error, which is expected.

**Comment by alisdair** [comment]
@RDK90 This crash looks like it may have been fixed in #22597. Could you please upgrade to the latest version of Terraform and see if the issue has been resolved?

**Comment by RDK90** [comment]
@alisdair I've upgraded to the v0.12.23 and it appears to be working now. Thanks for your help
